src/services/datasource.ts

```diff
@@ -50,35 +70,151 @@ class WrappedLokiDatasource extends RuntimeDataSource<DataQuery> {
             throw new Error('Invalid datasource!');
           }
 
-          const requestType = request.targets?.[0]?.resource;
+          const dataQueryRequest = { ...request };
+          // override the target datasource to Loki
+          dataQueryRequest.targets = request.targets.map((target) => {
+            target.datasource = ds;
+            return target;
+          });
+
+          dataQueryRequest.targets.forEach((target) => {
+            const requestType = target?.resource;
+
+            switch (requestType) {
+              case 'volume': {
+                this.getVolume(dataQueryRequest, ds, subscriber);
+                break;
+              }
+              case 'patterns': {
+                this.getPatterns(dataQueryRequest, ds, subscriber);
+                break;
+              }
+              default: {
+                this.getData(dataQueryRequest, ds, subscriber);
+                break;
+              }
+            }
+          });
+        });
+    });
+  }
+
+  private getData(
+    request: SceneDataQueryRequest,
+    ds: DataSourceWithBackend<DataQuery>,
+    subscriber: Subscriber<DataQueryResponse>
+  ) {
+    const dataQueryRequest = { ...request };
+    // override the target datasource to Loki
+    dataQueryRequest.targets = request.targets.filter((target) => {
+      return !target.resource;
+    });
+
+    if (!dataQueryRequest.targets.length) {
+      throw new Error('No valid queries!');
+    }
```

Comment:
```suggestion
```

Also, this can't really happen as we call it when we loop through `dataQueryRequest.targets`.